app/code/Magento/InventoryInStorePickupQuote/Plugin/Quote/ReplaceShippingAddressForShippingAddressManagement.php

```diff
@@ -0,0 +1,142 @@
+use Magento\InventoryInStorePickupApi\Api\GetPickupLocationInterface;
+use Magento\InventoryInStorePickupQuote\Model\IsPickupLocationShippingAddress;
+use Magento\InventoryInStorePickupQuote\Model\ToQuoteAddress;
+use Magento\InventoryInStorePickupShippingApi\Model\Carrier\InStorePickup;
+use Magento\InventorySalesApi\Api\Data\SalesChannelInterface;
+use Magento\Quote\Api\CartRepositoryInterface;
+use Magento\Quote\Api\Data\AddressInterface;
+use Magento\Quote\Api\Data\ShippingAssignmentInterface;
+use Magento\Quote\Model\ShippingAddressManagementInterface;
+use Magento\Store\Api\StoreRepositoryInterface;
+use Magento\Store\Api\WebsiteRepositoryInterface;
+
+/**
+ * Replace Shipping Address with Pickup Location Shipping Address for Shipping Address Management service.
+ */
+class ReplaceShippingAddressForShippingAddressManagement
+{
+    /**
+     * @var CartRepositoryInterface
+     */
+    private $cartRepository;
+
+    /**
+     * @var StoreRepositoryInterface
+     */
+    private $storeRepository;
+
+    /**
+     * @var WebsiteRepositoryInterface
+     */
+    private $websiteRepository;
+
+    /**
+     * @var GetPickupLocationInterface
+     */
+    private $getPickupLocation;
+
+    /**
+     * @var IsPickupLocationShippingAddress
+     */
+    private $isPickupLocationShippingAddress;
+
+    /**
+     * @var ToQuoteAddress
+     */
+    private $addressConverter;
+
+    /**
+     * @param CartRepositoryInterface $cartRepository
+     * @param StoreRepositoryInterface $storeRepository
+     * @param WebsiteRepositoryInterface $websiteRepository
+     * @param GetPickupLocationInterface $getPickupLocation
+     * @param IsPickupLocationShippingAddress $isPickupLocationShippingAddress
+     * @param ToQuoteAddress $addressConverter
+     */
+    public function __construct(
+        CartRepositoryInterface $cartRepository,
+        StoreRepositoryInterface $storeRepository,
+        WebsiteRepositoryInterface $websiteRepository,
+        GetPickupLocationInterface $getPickupLocation,
+        IsPickupLocationShippingAddress $isPickupLocationShippingAddress,
+        ToQuoteAddress $addressConverter
+    ) {
+        $this->cartRepository = $cartRepository;
+        $this->storeRepository = $storeRepository;
+        $this->websiteRepository = $websiteRepository;
+        $this->getPickupLocation = $getPickupLocation;
+        $this->isPickupLocationShippingAddress = $isPickupLocationShippingAddress;
+        $this->addressConverter = $addressConverter;
+    }
+
+    /**
+     * @param ShippingAddressManagementInterface $subject
+     * @param int $cartId
+     * @param AddressInterface $address
+     *
+     * @return array
+     * @throws NoSuchEntityException
+     * @throws StateException
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function beforeAssign(
+        ShippingAddressManagementInterface $subject,
+        int $cartId,
+        AddressInterface $address
+    ): array {
```

Comment:
The method is pretty overloaded, don't think? Please consider to move logic to the service and divide it into the logical parts.

Reply:
Any suggestions?

Reply:
We can set up the call to discuss this.

Reply:
Discussed: refactored.
Please review.